experimental/framework/tinylicious-client/src/test/TinyliciousClient.spec.ts

```diff
@@ -5,25 +5,142 @@
+            return true;
+        };
+
+        await assert.rejects(
+            containerAndServices,
+            errorFn,
+            "TinyliciousClient can get a container without initialization",
+        );
+    });
+
+    /**
+    * Scenario: test if TinyliciousClient can create a container without being initialized.
+    *
+    * Expected behavior: an error should be thrown when trying to create a container
+    * without initializing TinyliciousClient.
+    */
+    it("create a container without initializing", async () => {
+        const containerConfig: TinyliciousContainerConfig = { id: documentId };
+        const schema: ContainerSchema = {
+            name: documentId,
+            initialObjects: {
+                map1: SharedMap,
+            },
+        };
+        const containerAndServices = TinyliciousClient.createContainer(containerConfig, schema);
+        const errorFn = (error) => {
+            assert.notStrictEqual(error.message, undefined, "TinyliciousClient error is undefined");
+            return true;
+        };
+
+        await assert.rejects(
+            containerAndServices,
+            errorFn,
+            "TinyliciousClient can create a container without initialization",
+        );
+    });
+});
+
+describe("TinyliciousClient Post-Initialization", () => {
     before(() => {
-        const clientConfig: TinyliciousConnectionConfig = { port: 7070 };
-        TinyliciousClient.init(clientConfig);
+        TinyliciousClient.init();
     });
 
     let documentId: string;
+    const clientConfig: TinyliciousConnectionConfig = { port: 7070 };
     beforeEach(() => {
         documentId = uuid();
     });
 
+    /**
+    * Scenario: test if TinyliciousClient can get a non-exiting container.
+    *
+    * Expected behavior: an error should be thrown when trying to get a non-exisitng container.
+    */
+    it("load improperly created container (load a non-existing container)", async () => {
+        const containerConfig: TinyliciousContainerConfig = { id: documentId };
+        const schema: ContainerSchema = {
+            name: documentId,
+            initialObjects: {
+                map1: SharedMap,
+            },
+        };
+        const containerAndServices = TinyliciousClient.getContainer(containerConfig, schema);
+        const errorFn = (error) => {
+            assert.notStrictEqual(error.message, undefined, "TinyliciousClient error is undefined");
+            return true;
+        };
+
+        await assert.rejects(
+            containerAndServices,
+            errorFn,
+            "TinyliciousClient can load a non-existing container",
+        );
+    });
+
+    /**
+    * Scenario: test if TinyliciousClient can be initialized with a port number specified.
+    *
+    * Expected behavior: returned container should not be undefined.
+    */
+    it("create a container successfully with port number specification", async () => {
+        TinyliciousClient.init(clientConfig);
```

Comment:
this is essentially doing what the later test is doing since the client is a static that's already been initialized at this point.  if you want to test init with and without the clientConfig, they need to be in different test contexts (not sure how mocha handles that with globals), or you need to reset the global somehow

Reply:
Noting it down here per our discussion. Created a `ResettableTinyliciousClient` class as a child class of `TinyliciousClient` with a static method that resets the global instance. Also changed the global variable to `protected`. With this change, we can now reset/re-initialize the TinyliciousClient global variable if need be. To this end, developers are now exposed to the ability to do the same if they create a TinyliciousClient child class.